sdk/formrecognizer/Azure.AI.FormRecognizer/src/RecognizeCustomFormsOperation.cs

```diff
@@ -42,7 +42,18 @@ public class RecognizeCustomFormsOperation : Operation<RecognizedFormCollection>
         public override string Id { get; }
 
         /// <inheritdoc/>
-        public override RecognizedFormCollection Value => OperationHelpers.GetValue(ref _value);
+        public override RecognizedFormCollection Value
+        {
+            get
+            {
+                if (HasCompleted && !HasValue)
+#pragma warning disable CA1065 // Do not raise exceptions in unexpected locations
+                    throw new InvalidOperationException("RecognizeCustomForms operation failed.");
```

Comment:
@pakrym, is this the correct exception to raise here?  If we're doing this elsewhere in the SDK, would it make sense to pull this functionality into OperationHelpers in Core?

Reply:
It might be worth it if we find a design that works for multiple SDKs

Reply:
I looked into our guidelines and .NET guidelines when making this change and from my understanding `InvalidOperationException` will be better as it represents that the model is not right and therefore the user can't access the data.
When the user wants the value directly from the GET method, then it makes sense to raise `RequestFailedException`.

As for making a change that will be generic and benefit other libraries, I could only find some examples in keyvault and even in the library they have different behaviors per Operation class. It depends on the "state" the service is, and because that seems too specific, I decided to only leave this in FR.

Happy to look into it more if needed

Reply:
That sounds good.  Thanks for the investigation @maririos!

Reply:
Can we provide more diagnostics with this exception? `RecognizeCustomForms operation failed` is not very actionable.

Throwing RequestFailedException would allow us to at least include the final service response.